chef/cookbooks/provisioner/recipes/base.rb

```diff
@@ -328,13 +326,25 @@
       if current_priority != attrs[:priority]
         `zypper --non-interactive modifyrepo --priority #{attrs[:priority]} #{name}`
       end
+      # check if attrs contains the :purge key with the databag name as a value
+      if attrs[:purge]
+        Chef::Log.info("Purging no longer used #{name} zypper repository...")
+        `zypper --non-interactive removerepo #{name}`
+      end
     end
     # install additional packages
     os = "#{node[:platform]}-#{node[:platform_version]}"
     if node[:provisioner][:packages][os]
       node[:provisioner][:packages][os].each { |p| package p }
     end
   end
+elsif node[:platform_family] == "suse" && \
+    node.roles.include?("provisioner-server") && \
+    Provisioner::Repositories.all_repos_purged?(
+      node[:platform], node[:platform_version], node[:kernel][:machine]
+    )
```

Comment:
So that will only remove purged items related to the OS of the admin server. Don't you want to remove purged items for all OS?